scalafix-sbt/src/main/scala/scalafix/internal/sbt/ScalafixCompletions.scala

```diff
@@ -3,62 +3,167 @@ package scalafix.internal.sbt
+      .map { f =>
+        toAbsolutePath(Paths.get(f), cwd).toString
+      }
   }
+  private def gitDiffParser(cwd: Path): P = {
+    val jgitCompletion = new JGitCompletion(cwd)
+    token(
+      NotQuoted,
+      TokenCompletions.fixed(
+        (seen, level) => {
+          val last20Commits =
+            jgitCompletion.last20Commits
+              .filter { case (_, sha1) => sha1.startsWith(seen) }
+              .zipWithIndex
+              .map {
+                case ((log, sha1), i) => {
+                  val j = i + 1
+                  val idx = if (j < 10) " " + j.toString else j.toString
+                  new Token(
+                    display = s"|$idx| $log",
+                    append = sha1.stripPrefix(seen))
+                }
+              }
+              .toSet
+
+          val branchesAndTags =
+            jgitCompletion.branchesAndTags
+              .filter(info => info.startsWith(seen))
+              .map(info =>
+                new Token(display = info, append = info.stripPrefix(seen)))
+              .toSet
 
-  private def fileRule(cwd: Path): Parser[String] =
-    token("file:") ~>
-      StringBasic
-        .examples(new AbsolutePathExamples(cwd))
-        .map { f =>
-          val path = toAbsolutePath(Paths.get(f), cwd).toString
-          "file:" + path
+          Completions.strict(last20Commits ++ branchesAndTags)
         }
+      )
+    )
+  }
 
-  private def uri(protocol: String) =
-    token(protocol + ":") ~> NotQuoted.map(x => s"$protocol:$x")
+  def parser(cwd: Path, compat: Boolean): Parser[Seq[String]] = {
+    val pathParser: P = token(filepathParser(cwd))
+    val pathRegexParser: P = mapOrFail(pathParser) { regex =>
+      Pattern.compile(regex); regex
+    }
+    val classpathParser: P = repsep(pathParser, File.pathSeparator)
+    val fileRule: P = (token("file:") ~ pathParser.map("file:" + _)).map {
+      case a ~ b => a + b
+    }
+    val ruleParser = token(
+      namedRule | fileRule | uri("github") | uri("replace") | uri("http") | uri(
+        "https") | uri("scala"))
 
-  private val namedRule: Parser[String] =
-    names.map(literal).reduceLeft(_ | _)
-
-  def parser(cwd: Path): Parser[Seq[String]] = {
-    val all =
-      namedRule |
-        fileRule(cwd) |
-        uri("github") |
-        uri("replace") |
-        uri("http") |
-        uri("https") |
-        uri("scala")
-    (token(Space) ~> token(all)).* <~ SpaceClass.*
+    val bash: P = "--bash"
+    val classpath: P = arg("--classpath", classpathParser)
+    val classpathAutoRoots: P = arg("--classpath-auto-roots", classpathParser)
+    val config: P = arg("--config", pathParser)
+    val configStr: P = arg("--config-str", string.examples())
+    val diff: P = "--diff"
+    val diffBase: P = arg("--diff-base", gitDiffParser(cwd))
+    val exclude: P = arg("--exclude", pathRegexParser)
+    val noStrictSemanticdb: P = "--no-strict-semanticdb"
+    val noSysExit: P = "--no-sys-exit"
```

Comment:
This needs to be always enabled from sbt in order to not exit the shell? The first thing I tried while testing this PR  locally was `> scalafix` and that killed the sbt shell session.

Reply:
Unchanged: https://github.com/MasseGuillaume/scalafix/blob/4511dd57e7702b9796cb92deb138ef3e5b612348/scalafix-sbt/src/main/scala/scalafix/sbt/ScalafixPlugin.scala#L274